openshift/machine-config-operator
# MCO-573: Wire up security/trust/pull secrets between rpm-ostree and internal registry

- Added a new service account for doing image pulls from the internal registry
- Added a new field in ControllerConfig for the secret generated from this new SA
- The secret is then merged with the global pull secret before being applied to the machine config, which templates it out to a file.
- The daemon now first attempts to link rpm-ostree's auth file to the new merged secret file(as long as it exists and not empty, otherwise it defaults to global pull secret). 
- Added a rebootless carveout for the "merged" secrets file so we do not reboot if the secret changes

This does not work for scaling up nodes - this only allows access for nodes that are already in the cluster.  [MCO-578](https://issues.redhat.com/browse/MCO-578) is intended for that. I did add a default route path in the image registry secret file, but solving the scale up problem needs some additional certs finesse. 

I've split out the commits for easier review, but I can squash if necessary. A lot of this was based on John's work from #3397. Thanks a bunch, @jkyros! 

How to test:
(i) [Load](https://docs.openshift.com/container-platform/4.13/registry/accessing-the-registry.html#registry-accessing-directly_accessing-the-registry) an RHCOS image into the MCO's slice of the image registry via podman/docker. 
(ii) Apply an MC to an infra node with an overriden osImageURL that points to this new image in the image registry
(iii) The infra node should successfully rebase to the new image(observe logs on the node)
(iv) The global pull secret should still work; this can be tested by rolling the node back to the original MC.
